### ISSUE TITLE
prompt parameter cannot be overridden

### ISSUE THREAD

**Issue by martinlingstuyl** [AzureAD/azure-activedirectory-library-for-cordova#167]
### Issue Description
I need to override the prompt parameter with prompt=consent because of an issue with AAD that can be reproduced like this:
1) Create a Native AAD app and give it some permissions
2) Use the AAD app in cordova > sign in, a consent screen is shown.
3) Now add a new permission scope to the Native AAD app.
4) Try to sign in again: it will not show a consent screen after signin in and thus no token can be requested for the specific new resource, because no consent was given by the user for that resource. 

I get it with ADAL JS as well. Though there I can force prompt=consent by adding it to extraQueryParameters variable.

In ADAL for cordova I cannot add prompt=consent to extraQueryParameters. It will throw a duplicate parameter message.

### Repro Environment Details
1. Installed plugin version:  0.8.x and 0.10.x
2. Target Platform: tested on ios & windows. Didn't test android.
3. Cordova and platform versions: 5.4.1
4. Test device and OS version: Windows 10 desktop

**Comment by matrosov-nikita** [comment]
@martinlingstuyl Thanks so much for taking the time to submit this report! I can confirm using the latest `ADAL@0.10.1` I am able to reproduce this issue.

It looks like a bug in the .NET library caused due to adding `prompt=login` by default to extra query parameters. The only workaround I found now is to use `PromptBehaviour.auto` as platform parameter.

Could you please verify this assumption by replacing `Microsoft.IdentityModel.Clients.ActiveDirectory.PromptBehavior.always` with `Microsoft.IdentityModel.Clients.ActiveDirectory.PromptBehavior.auto` on this lines:

If you're running on SSO mode - https://github.com/AzureAD/azure-activedirectory-library-for-cordova/blob/master/src/windows/ADALProxy.js#L182
else - https://github.com/AzureAD/azure-activedirectory-library-for-cordova/blob/master/src/windows/ADALProxy.js#L188

You can find plugin's source code in the `<PROJECT_PATH>\platforms\windows\www\plugins\cordova-plugin-ms-adal\src\windows\ADALProxy.js`

**Comment by martinlingstuyl** [comment]
Hi Nikita,

Thanks for responding!

This does indeed 'fix' the issue on the windows platform. When changing the PromptBehavior parameter to 'auto', I can see the prompt parameter disappearing from the Azure AD login dialog url. If I then add prompt=consent to extraQueryParameters, this is correctly appended to the login dialog url, and it shows a nice consent screen after signing in.

Of course, you guys added the PromptBehavior.always on purpose, so changing it is no option. But possibly you could update the code (of all the platforms) so it contains a check for a possible prompt override.

**Comment by tobbbe** [comment]
@martinlingstuyl @matrosov-nikita I tried changing https://github.com/AzureAD/azure-activedirectory-library-for-cordova/blob/212ecda23c0e454083fcc236117b4c8e51630d69/src/android/CordovaAdalPlugin.java#L42 to `Microsoft.IdentityModel.Clients.ActiveDirectory.PromptBehavior.Auto`.

But when I try to login again after login out (https://github.com/AzureAD/azure-activedirectory-library-for-cordova#how-to-sign-out) I can choose and login the account I just logged out from if I just click (no password)?!
![image](https://user-images.githubusercontent.com/2511610/32780543-f43d1682-c941-11e7-94fe-dd35b22fcac9.png)

Also, is there no other way to prompt users to accept permissions when I have added new permissions?

**Comment by oliverjanik** [comment]
Any news on this? Why is the default behaviour so bad?  It makes no sense. What use-case is there for letting a user through with no prompt only to get 403 when they request a resource?

I want the users to see the consent screen every time the permissions change. How can I do that?

@matrosov-nikita

**Comment by martinlingstuyl** [comment]
@oliverjanik @tobbbe : I agree with you guys that this is strange behavior. I don't know why MS didn't change this yet. My guess is they're primarily focussing on MSAL although they say it's not a substitute for ADAL. (and there is no cordova plugin for that unfortunately.)

Our way to deal with it was to simply swap the AD application for a new one with the updated permissions. Not a very good solution, but alas, what can one do.

**Comment by martinlingstuyl** [comment]
Hi guys, 

I've added a fix for this issue. Hope MS will pull my commit, @matrosov-nikita .

@oliverjanik: you can always use my fork or copy the code if you want.
